### ISSUE TITLE
Add an ability to specify Page Header and Footers through Org snippets

### ISSUE THREAD

**Issue by kjambunathan** [kjambunathan/org-mode-ox-odt#221]

**Comment by kjambunathan** [comment]
Here is a test case for this feature 

To achieve this

![pageheaderfooter](https://user-images.githubusercontent.com/19869864/205558404-b2386716-feb9-41d0-97c1-3940b2cfd8fd.png)


do this



```
#+ODT_DOCUMENT_PROPERTIES: DOC-TITLE DOC-VERSION

#+DOC-TITLE: How to customize Page Headers and Footers
#+DOC-VERSION: 1

#+Macro: DocTitle {{{keyword-to-documentproperty(DOC-TITLE)}}}
#+MACRO: DocVersion {{{keyword-to-documentproperty(DOC-VERSION)}}}

This paragraph is in a ~Standard~ page.  It has a header.

- The header is an image.

#+ATTR_ODT: :page-style "Standard"
#+begin_center
    #+begin_header
        #+ATTR_ODT: :height 3 :inner-frame  (anchor "as-char")
        [[./org-mode-unicorn.png]]
    #+end_header
#+end_center

#+begin_center
--- *END OF PAGE* ---
#+end_center

#+ATTR_ODT: :page-break "before" :page-style "OrgPage"
This paragraph is in an ~OrgPage~. It has a /different/ header and a
footer.

- The header is ~Tab~-separated list of ~OpenDocument fields~.  It
  carries ~Document Title~.

- The footer is an ~Org Table~ with ~User-defined fields~

#+ATTR_ODT: :page-style "OrgPage"
#+begin_header
    #+ATTR_ODT: :style "Header"
    {{{ODTTab}}}{{{DocTitle}}}{{{ODTTab}}}
#+end_header

#+ATTR_ODT: :page-style "OrgPage"
#+begin_footer
    #+ATTR_ODT: :col-cookies "||c||"    
    #+ATTR_ODT: :widths "1,2,1"
    |   | {{{DocTitle}}} ver. {{{DocVersion}}}      |   |
    |   | {{{ODTPageNumber}}} of {{{ODTPageCount}}} |   |
#+end_footer

#+begin_center
--- *END OF PAGE* ---
#+end_center

#+ATTR_ODT: :page-break "before" :page-style "OrgFrontMatterPage"
This paragraph is in an ~OrgFrontMatterPage~.

- The footer is /suppressed/. FYI, by default, ~OrgFrontMatterPage~
  uses lower-case Roman numerals.

#+ATTR_ODT: :page-style "OrgFrontMatterPage"
#+begin_footer
#+end_footer

#+begin_center
--- *END OF PAGE* ---
#+end_center
```

**Comment by kjambunathan** [comment]
@PiotrPanasiuk,

Use this page for discussing the new page header / footer feature.  There is too much detail in the old bug reports (which are obsoleted now).

So, add your questions, comments, requests here..


I noticed that your are using ODF 1.3 files.  Any specific reason you use ODF 1.3.  The OpenDocument Exporter supports ony ODF 1.2.    See https://github.com/kjambunathan/org-mode-ox-odt/discussions/170#discussioncomment-2363953 if you want to save your ODT files as ODF 1.2.

**Comment by kjambunathan** [comment]
@PiotrPanasiuk

Please pull again before testing ... 

I had some commits in my workarea which I forgot to push.  I have pushed it now.

**Comment by PiotrPanasiuk** [comment]
version which i have from repo is "ox-odt 9.5.3.470" and spacemacs report's me "Wrong type argument: stringp, nil". Do You suspect what could be wrong?

**Comment by kjambunathan** [comment]
> version which i have from repo is "ox-odt 9.5.3.470" and spacemacs report's me "Wrong type argument: stringp, nil". Do You suspect what could be wrong?



This is a bug.  I have seen this happen when a linked inline image doesn't exist.   In other words, if you are copy-pasting my test case for verifying this feature you aren't having `org-mode-unicorn.png' in the local directory.  Just put any image you want there.  

I circulated a modified zip file https://github.com/kjambunathan/org-mode-ox-odt/files/10151235/document_template_modified.zip as part of comment https://github.com/kjambunathan/org-mode-ox-odt/issues/182#issuecomment-1336826686.

Did you face any issue exporting the `org' file in there.

----

I just released `ox-odt-9.6.524.tar'.  This doesn't have the stringp fix ... so ensure that all the linked inline images actually exists in that path.  I will open a bug to track the `stringp' issue.

**Comment by kjambunathan** [comment]
> version which i have from repo is "ox-odt 9.5.3.470" and spacemacs report's me "Wrong type argument: stringp, nil". Do You suspect what could be wrong?

This `stringp` error is fixed in `ox-odt-9.6.525.tar`.  Please upgrade to this version.  Remember to upgrade `org` to `9.6` before upgrading the `ox-odt`.  `org-9.6` was released a couple of weeks ago.
